OMEdit/OMEditGUI/Git/RevertCommitsDialog.cpp

```diff
@@ -0,0 +1,264 @@
+  QFileInfo fileInfo(repository);
+  QString directory = fileInfo.absoluteDir().absolutePath();
+  mpProcess->setWorkingDirectory(directory);
+  mpProcess->start("git", { "revert", mpCommitTextBox->text() });
+  mpProcess->waitForFinished();
+  QString revert =  mpProcess->readAllStandardOutput();
+  accept();
+}
+
+void RevertCommitsDialog::browseCommitLog()
+{
+  LogCommitDialog *pLogCommitDialog = new LogCommitDialog(this);
+  pLogCommitDialog->exec();
+}
+
+void RevertCommitsDialog::workingDirectoryChanged(const QString &workingDirectory)
+{
+
+}
+
+void RevertCommitsDialog::browseWorkingDirectory()
+{
+  QString workingDirectory = StringHandler::getExistingDirectory(this, QString("%1 - %2").arg(Helper::applicationName).arg(Helper::chooseDirectory), NULL);
+  if (workingDirectory.isEmpty())
+    return;
+  mpWorkingDirectoryTextBox->setText(workingDirectory);
+}
+
+void RevertCommitsDialog::setCommit(QString commit)
+{
+  mpCommitTextBox->setText(commit);
+}
+/*!
+ * \class LogCommitDialog
+ * \brief Creates a dialog that shows commit log.
+ */
+/*!
+ * \brief LogCommitDialog::LogCommitDialog
+ * \param pMainWindow - pointer to MainWindow
+ */
+
+enum Columns
+{
+    Sha1Column,
+    SubjectColumn,
+    ColumnCount
+};
+
+LogCommitDialog::LogCommitDialog(RevertCommitsDialog *pRevertCommitsDialog)
+  : QDialog(pRevertCommitsDialog)
+{
+  setWindowTitle(QString(Helper::applicationName).append(" - ").append("Select Commit"));
+  setAttribute(Qt::WA_DeleteOnClose);
+  resize(500, 400);
+  mpRevertCommitsDialog = pRevertCommitsDialog;
+  mpLogModel = new QStandardItemModel(0, ColumnCount, this);
+  QStringList headers;
+  headers << tr("Sha1")<< tr("Subject");
+  mpLogModel->setHorizontalHeaderLabels(headers);
+  // Log commit tree view
+  mpLogTreeView = new QTreeView;
+  mpLogTreeView->setModel(mpLogModel);
+  mpLogTreeView->setMinimumWidth(300);
+  mpLogTreeView->setUniformRowHeights(true);
+  mpLogTreeView->setRootIsDecorated(false);
+  mpLogTreeView->setSelectionBehavior(QAbstractItemView::SelectRows);
+  mpLogTreeView->setSelectionMode(QAbstractItemView::SingleSelection);
+  // Create the buttons
+  mpOkButton = new QPushButton(tr("Ok"));
+  mpOkButton->setEnabled(true);
+  connect(mpOkButton, SIGNAL(clicked()), SLOT(ok()));
+  mpCancelButton = new QPushButton(Helper::cancel);
+  connect(mpCancelButton, SIGNAL(clicked()), SLOT(reject()));
+  // create buttons box
+  mpButtonBox = new QDialogButtonBox(Qt::Horizontal);
+  mpButtonBox->addButton(mpOkButton, QDialogButtonBox::ActionRole);
+  mpButtonBox->addButton(mpCancelButton, QDialogButtonBox::ActionRole);
+  // set the layout
+  QGridLayout *pMainLayout = new QGridLayout;
+  pMainLayout->setAlignment(Qt::AlignTop | Qt::AlignLeft);
+  pMainLayout->addWidget(mpLogTreeView, 0, 0);
+  pMainLayout->addWidget(mpButtonBox, 1, 0);
+  setLayout(pMainLayout);
+
+  getCommitLog();
+}
+
+void LogCommitDialog::getCommitLog()
+{
+  mpProcess = new QProcess(this);
```

Comment:
You need to call `delete mpProcess` after you are done with other. Otherwise we have a memory leak.
Better to use QProcess process; with allocating memory on the heap.